core/lib/types/src/api/mod.rs

```diff
@@ -19,6 +19,14 @@ use crate::{
 
 pub mod en;
 
+/// Enum for choosing API mode
+#[derive(Copy, Clone, Debug, PartialEq, Display, Default)]
+pub enum APIMode {
```

Comment:
Bikeshedding: [Rust guidelines recommend](https://rust-lang.github.io/rust-clippy/master/index.html#/upper_case_acronyms) not capitalizing acronyms; i.e., it should be `ApiMode`.